### ISSUE TITLE
implementing a simple LSTM

### ISSUE THREAD

**Issue by xcapt** [tensorflow/tfjs#752]
Hello,
Based on the addition RNN sample, I am trying to implement a very simple LSTM based, that would take  a sequence of 0 and 1 and output the AND value.

`var xs = tf.tensor3d([[[1], [1]], [[1], [0]], [[0], [1]], [[0], [0]]]);
var ys = tf.tensor3d([[[1]], [[0]], [[0]], [[0]]]);

const maxLen = 2, vocabularySize = 1, hiddenSize = 3;

const model = tf.sequential();
model.add(tf.layers.lstm({
        units: hiddenSize,
        recurrentInitializer: 'glorotNormal',
        inputShape: [maxLen, vocabularySize]
      }));
model.add(tf.layers.repeatVector({ n: vocabularySize+1 }));
                        model.add(tf.layers.lstm({
        units: hiddenSize,
        recurrentInitializer: 'glorotNormal',
        returnSequences: true
      }));
model.add(tf.layers.timeDistributed({ layer: tf.layers.dense({ units: vocabularySize }) }));
model.summary();
model.compile({loss: tf.losses.meanSquaredError, optimizer: 'sgd'})
model.fit(xs, ys, {
                                        epochs:100,                                    
                                        }).then((history) => {
                          // Use the model to do inference on a data point the model hasn't seen before:
                          // Open the browser devtools to see the output
                          model.predict(tf.tensor2d([1,0],[1,2])).print();
                          model.predict(tf.tensor2d([0,0],[1,2])).print();
                          model.predict(tf.tensor2d([0,1],[1,2])).print();
                          model.predict(tf.tensor2d([1,1],[1,2])).print();        
                        });
`

However I get

`Error when checking target: expected time_distributed_TimeDistributed1 to have shape [,2,1], but got array with shape [4,1,1]
`

I tried various combinations based on the examples but cant find the issue.

Any one could give me a hint ?

Thank you

**Comment by xcapt** [comment]
mulitple rookie mistakes

working code here:
[](https://codepen.io/xcapt/pen/qJbrEq)
